core/server/common/src/main/java/alluxio/master/journal/raft/JournalStateMachine.java

```diff
@@ -278,48 +253,32 @@ public long takeSnapshot() {
     long index;
     StateLockManager stateLockManager = mStateLockManagerRef.get();
     if (!mIsLeader) {
+      LOG.debug("Taking local snapshot as follower");
       index = takeLocalSnapshot(false);
     } else if (stateLockManager != null) {
       // the leader has been allowed to take a local snapshot by being given a non-null
       // StateLockManager through the #allowLeaderSnapshots method
       try (LockResource stateLock = stateLockManager.lockExclusive(StateLockOptions.defaults())) {
+        LOG.debug("Taking local snapshot as leader");
         index = takeLocalSnapshot(true);
       } catch (Exception e) {
         return RaftLog.INVALID_LOG_INDEX;
       }
     } else {
-      RaftGroup group;
-      try (LockResource ignored = new LockResource(mGroupLock)) {
-        if (mServerClosing) {
-          return RaftLog.INVALID_LOG_INDEX;
-        }
-        // These calls are protected by mGroupLock and mServerClosing
-        // as they will access the lock in RaftServerProxy.java
-        // which is also accessed during raft server shutdown which
-        // can cause a deadlock as the shutdown takes the lock while
-        // waiting for this thread to finish
-        Preconditions.checkState(mServer.getGroups().iterator().hasNext());
-        group = mServer.getGroups().iterator().next();
-      } catch (IOException e) {
-        SAMPLING_LOG.warn("Failed to get raft group info: {}", e.getMessage());
-        return RaftLog.INVALID_LOG_INDEX;
-      }
-      if (group.getPeers().size() < 2) {
-        SAMPLING_LOG.warn("No follower to perform delegated snapshot. Please add more masters to "
-            + "the quorum or manually take snapshot using 'alluxio fsadmin journal checkpoint'");
-        return RaftLog.INVALID_LOG_INDEX;
-      } else {
-        index = mSnapshotManager.maybeCopySnapshotFromFollower();
-      }
+      index = mSnapshotManager.downloadSnapshotFromOtherMasters();
     }
     // update metrics if took a snapshot
     if (index != RaftLog.INVALID_LOG_INDEX) {
       mSnapshotLastIndex = index;
-      mLastCheckPointTime = System.currentTimeMillis();
+      mLastSnapshotTime = System.currentTimeMillis();
+      LOG.info("Took snapshot up to index {} at time {}", mSnapshotLastIndex, DateTime.now());
```

Comment:
nit: im actually not sure what timezone this `DateTime.now()` is. You may wanna double check it doesn't default to UTC. 
```
DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT).ofPattern(“yyyyMMdd-HHmmss”)
                  .withLocale(Locale.getDefault()).withZone(ZoneId.systemDefault());
// This gives you a date time in your zone
formatter.format(Instant.now());
```

Reply:
It does not default to UTC.